File: server/src/main/java/io/github/hulang1024/chinesechess/user/SearchUserInfo.java
package io.github.hulang1024.chinesechess.user;

import com.alibaba.fastjson.annotation.JSONField;
import io.github.hulang1024.chinesechess.userstats.UserStats;
import lombok.Data;

@Data
public class SearchUserInfo extends User {
    private Boolean isOnline;
    private Boolean isFriend;
    private Boolean isMutual;
    private UserStats userStats;
    @JSONField(serialize = false)
    private UserStatus status;

    private String loginDeviceOS;

    public SearchUserInfo(User user) {
        setId(user.getId());
        setNickname(user.getNickname());
        setAvatarUrl(user.getAvatarUrl());
    }

    @JSONField(name = "status")
    public int getStatusCode() {
        return status == null ? -1 : status.getCode();
    }
}

File: client/src/online/user/SearchUserInfo.ts
import User from "src/user/User";
import UserStats from "../../user/UserStats";
import UserStatus from "../../user/UserStatus";
import UserDeviceInfo from "./UserDeviceInfo";

export default class SearchUserInfo extends User {
  isOnline: boolean;

  isFriend: boolean;

  isMutual: boolean;

  userStats: UserStats;

  status: UserStatus;

  deviceInfo: UserDeviceInfo;
}
